src/main/java/teho/main/TehO.java

```diff
@@ -0,0 +1,199 @@
+        task.markAsDone(taskNumber);
+    }
+
+    /**
+     * Represents the unmarking of task according to user's command.
+     *
+     * @param userCommand User's command containing the task to be unmarked.
+     */
+    public void unmarkTask(String userCommand) {
+        int taskNumber = parseInt(userCommand.split(" ")[1]) - 1; //counting from 0
+        Task task = this.taskList.getTask(taskNumber);
+        task.markAsNotDone(taskNumber);
+    }
+
+    /**
+     * Represents the adding of new ToDo task into taskList according to user's command.
+     *
+     * @param userCommand User's command containing the new ToDo task to be added into taskList.
+     */
+    public String addToDo(String userCommand) {
+        try {
+            if (userCommand.length() < 5) {
+                throw new EmptyToDoDescriptionException();
+            }
+            String command = userCommand.substring(5); //"todo " 5 index
+            Task task = new ToDo(command);
+            this.taskList.add(task);
+            return ui.generateAddToDoMessage(task, taskList);
+        } catch (EmptyToDoDescriptionException e) {
+            return e.toString();
+        }
+    }
+
+    /**
+     * Represents the adding of new Deadline task into taskList according to user's command.
+     *
+     * @param userCommand User's command containing the new Deadline task to be added into taskList.
+     */
+    public  String addDeadline(String userCommand) {
+        try {
+            if (userCommand.length() < 9) {
+                throw new EmptyDeadlineDescriptionException();
+            }
+            String commandWithDate = userCommand.substring(9); //"deadline " 9 index
+            String cDeadline = commandWithDate.split(" /by ")[0]; //just description
+            LocalDate byDate = LocalDate.parse(commandWithDate.split(" /by ")[1]); //just byDate
+            Task task = new Deadline(cDeadline, byDate);
+            this.taskList.add(task);
+            return ui.generateAddDeadlineMessage(task, taskList);
+        } catch (EmptyDeadlineDescriptionException e) {
+            return e.toString();
+        }
+    }
+
+    /**
+     * Represents the addition of new Event task into taskList according to user's command.
+     *
+     * @param userCommand User's command containing the new Event task to be added into taskList.
+     */
+    public String addEvent(String userCommand) {
+        try {
+            if (userCommand.length() < 6) {
+                throw new EmptyEventDescriptionException();
+            }
+            String commandWithDate = userCommand.substring(6); //"todo " 6 index
+            String cEvent = commandWithDate.split(" /from ")[0]; //just description
+            String dates = commandWithDate.split(" /from ")[1]; //bothDates
+            LocalDate fromDate = LocalDate.parse(dates.split(" /to ")[0]); //just fromDate
+            LocalDate toDate = LocalDate.parse(dates.split(" /to ")[1]); //just toDate
```

Comment:
I like how clear the variable naming is (so much so that the comments are a bit unecessary).